lib/TokenManager.ts

```diff
@@ -166,8 +177,8 @@ function renew(sdk, tokenMgmtRef, storage, key) {
         // renewed token, we verify the promise key doesn't exist and return.
         return;
       }
-      add(sdk, tokenMgmtRef, storage, key, freshToken);
-      tokenMgmtRef.emitter.emit('renewed', key, freshToken, oldToken);
+      add(sdk, tokenMgmtRef, storage, key, freshToken, false /* shouldEmitEvent */);
```

Comment:
What is the issue with emitting an "added" event for fresh tokens obtained from a renew operation?

Reply:
`add` will be called in `renew` process (put new token into storage), which will end up with two events be emitted in one event, which may then cause cascading re-evaluation. It would be better to let TokenManager make the decision to only emit events that make sense to the downstream.

Reply:
Can you elaborate on the "cascading re-evaluation"? I don't think TokenManager could possibly know what makes sense to the downstream. If a successful renew event always creates 2 events, then the downstream can listen to either one. "added"/"removed" events would be important if someone wanted to synchronize with the storage. "renewed/error" events have a different meaning, which may not be as useful. In general I  don't think we should suppress events even if they feel redundant.

Reply:
I agree, TokenManager should expose all events to the downstream. Actually, the logic should be handled in `AuthStateManager` (the downstream) to only re-evaluate `authState` based on `added` and `removed`.